vars/skipStage.groovy

```diff
@@ -145,7 +145,7 @@ boolean skip_ftest(String distro, String target_branch) {
 
     // When the stage is started manually or via a timer skipping the stage is
     // controlled by the build parameter and commit pragmas are ignored
-    if (startedByTimer() || startedByUser()) {
+    if ((startedByTimer() || startedByUser()) && !startedByUpstream()) {
```

Comment:
Is `!` the condition we want here?  My  concern was that if an upstream job starts a downstream job from a PR (rather than it's default of `master` or `release/2.2` and that PR had commit pragmas, the upstream job has no ability to override them with the parameters it initiates the downstream job with.

It's a corner-case (or maybe rare is a better word -- that somebody uses a PR/branch for a downstream test job) to be sure, but it something that is currently possible with the given upstream/downstream testing framework that I have put into place.

Reply:
What commit message does a downstream job read when it is started by an upstream job?  Does it read the commit message from the upstream job?

Reply:
I've reworked this fix and its now the changes to `skip_ftest_hw` are modeled after theapproach used in `skip_ftest`.